garrysmod/gamemodes/base/entities/entities/lua_run.lua

```diff
@@ -1,10 +1,14 @@
 
 ENT.Type = "point"
 ENT.DisableDuplicator = true
+ENT.MapCreationID = -1
 
 AccessorFunc( ENT, "m_bDefaultCode", "DefaultCode" )
 
 function ENT:Initialize()
+
+	self.MapCreationID = self:MapCreationID()

```

Comment:
This overwrites `MapCreationID`, name it something else as it'll break any addon that attempts to call it. Also, if you are going to put this here, do what I suggested and put all of it here, e.g.

```lua
self.RunStringID = game.GetMap() .. ":" .. self:MapCreationID()
```